com.unity.shadergraph/Editor/ShaderFoundry/Linker/Properties/MaterialPropertyDeclarationData.cs

```diff
@@ -0,0 +1,53 @@
+using System.Collections.Generic;
+
+namespace UnityEditor.ShaderFoundry
+{
+    class MaterialPropertyDeclarationData
+    {
+        internal IEnumerable<string> Attributes;
+        internal string UniformName;
+        internal string DisplayName;
+        internal string DisplayType;
+        internal string DefaultValueExpression;
+
+        internal static MaterialPropertyDeclarationData BuildSimple(FieldPropertyContext context, string displayType, string defaultValue, FieldPropertyData resultProperty)
+        {
+            if (context.Exposed == false)
+                return null;
+
+            // Build the list of shaderlab attributes for the property
+            var attributes = new List<string>();
+            if (context.ExtraAttributes != null)
+                attributes.AddRange(context.ExtraAttributes);
+            foreach (var attribute in context.ProcessedAttributes.Values)
```

Comment:
are ProcessedAttributes intended to be a dictionary for quickly looking up attributes by name?  Or a list of the attributes that get pushed straight through to shaderlab?   If it's the second, we can use a List<> or a HashSet<> instead of a Dictionary<>

Reply:
I would also recommend changing the name, or at least adding a comment about what is recorded there.

Reply:
This part is definitely very premature, and will need improvements later, but the idea was that processed attributes were ones that have been verified to be valid attributes for forwarding to shaderlab. Not all attributes on a property are valid shaderlab attributes, so this filters them.

I'll think on a name and add a comment.

Reply:
We can (and should) revisit this later, but I'm leaving it as a dictionary because not all ShaderLab attributes make sense on all property types (e.g. "MainTexture" doesn't make sense on a float).